pkg/controllers/machine/disruption/controller.go

```diff
@@ -0,0 +1,153 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package disruption
+
+import (
+	"context"
+
+	"github.com/samber/lo"
+	"go.uber.org/multierr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/clock"
+	"knative.dev/pkg/logging"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/cloudprovider"
+	corecontroller "github.com/aws/karpenter-core/pkg/operator/controller"
+	"github.com/aws/karpenter-core/pkg/utils/result"
+)
+
+type machineReconciler interface {
+	Reconcile(context.Context, *v1alpha5.Provisioner, *v1alpha5.Machine) (reconcile.Result, error)
+}
+
+var _ corecontroller.TypedController[*v1alpha5.Machine] = (*Controller)(nil)
+
+// Controller is a disruption controller that adds StatusConditions to Machines when they meet the
+// disruption criteria for that disruption type
+// i.e. When the Machine has surpassed its owning provisioner's expirationTTL, then it is marked as "VoluntarilyDisrupted"
```

Comment:
Latin abbreviation nit: i.e. should be e.g. here

Reply:
Out of curiosity, why?

Reply:
OED says:

e.g. adv. for example, for instance.
 [Initialism < [exempli gratia adv.](https://www-oed-com.proxy.multcolib.org/view/Entry/66062#eid4994146) Similar use is found in Latin texts contemporary with quot. [1622](https://www-oed-com.proxy.multcolib.org/view/Entry/58919?redirectedFrom=e.g.#eid1006478261).]

> "I, n.1". OED Online. March 2023. Oxford University Press. https://www-oed-com.proxy.multcolib.org/view/Entry/90670?rskey=rjALrx&result=1&isAdvanced=false (accessed May 05, 2023).

Compare to i.e.:

i.e. n. id est (Latin), that is (to say).

> "E, n.1". OED Online. March 2023. Oxford University Press. https://www-oed-com.proxy.multcolib.org/view/Entry/58919?redirectedFrom=e.g. (accessed May 05, 2023).

I'm reading the code comment as "For example, when the machine has surpassed..." In which case e.g. is the abbreviation you want. You wanna use "i.e." in place of "in other words, ..."

If I read the comment wrong then ignore (too late to ignore, probably)

Reply:
Ah, I think I've been using that expression wrong my whole life 😄 Good to know